compiler/src/dotty/tools/dotc/Compiler.scala

```diff
@@ -87,7 +87,8 @@ class Compiler {
          new sjs.ExplicitJSClasses,  // Make all JS classes explicit (Scala.js only)
          new ExplicitOuter,          // Add accessors to outer classes from nested ones.
          new ExplicitSelf,           // Make references to non-trivial self types explicit as casts
-         new StringInterpolatorOpt) :: // Optimizes raw and s and f string interpolators by rewriting them to string concatenations or formats
+         new StringInterpolatorOpt,  // Optimizes raw and s and f string interpolators by rewriting them to string concatenations or formats
+         new DropBreaks) ::          // Optimize local Break throws by rewriting them 
```

Comment:
Since this is an optimization phase, I would expect it to go after Erasure. It should be more robust over there, and perhaps simpler. Is there a reason that it comes this early?

Reply:
I tried that first and backed out. The reason was that erasure inserts casts so we have deal with them instead. I can have a look again to see what can be done about it.

Reply:
Here's the first test in `breakOpt` after erasure:
```scala
[[syntax trees at end of                   erasure]] // test1.scala
package <empty> {
  final lazy module val breakOpt: breakOpt = new breakOpt()
  @SourceFile("test1.scala") final module class breakOpt() extends Object() {
    private def writeReplace(): Object =
      new scala.runtime.ModuleSerializationProxy(classOf[breakOpt])
    def zero(): Int = 0
    def zero_=(x$1: Int): Unit = ()
    def test1(x: Int): Int =
      {
        val local: scala.util.boundary.Label = new scala.util.boundary.Label()
        (try 
          {
            if x.<(0) then
              {
                val value$proxy1: Int = breakOpt.zero()
                local.break(scala.Int.box(value$proxy1))
              }
             else ()
            x
          }
         catch 
          {
            case ex @ _:scala.util.boundary.Break =>
              (if ex.label().eq(local) then scala.Int.unbox(ex.value()) else 
                throw ex):Int
          }
        )
      }:Int
  }
}
```
The problem is that we'd have to undo the box/unbox that's going on. That looks harder than dealing with Inlined.

Reply:
Hum, indeed, that is annoying. Let's keep it before erasure, then.